omegaconf/listconfig.py

```diff
@@ -234,6 +243,11 @@ def append(self, item: Any) -> None:
             index = len(self)
             self._validate_set(key=index, value=item)
 
+            element_type = self.__dict__["_metadata"].element_type
+            expect_container = is_container_annotation(element_type)
+            if expect_container and isinstance(item, BaseContainer):
+                item = item._value()
+
```

Comment:
I am  not seeing new tests for deepcopy/copy that are testing with containers as element types.
Can you add some?